src/app/content/components/Page/PageComponent.tsx

```diff
@@ -216,4 +221,26 @@ export default class PageComponent extends Component<PagePropTypes> {
 
     return promise;
   }
+
+  private getRunId(): number {
+    const newId = this.componentDidUpdateCounter + 1;
+    this.componentDidUpdateCounter = newId;
+    return newId;
+  }
+
+  /**
+   * When a user navigates quickly between pages there are multiple calls to componentDidUpdate
+   * and since it is an async function there might be still unresolved promisses that would result
+   * in calling highlighter.update multiple times, see: https://github.com/openstax/unified/issues/1169
+   */
+  private shouldUpdateHighlightManagers(prevProps: PagePropTypes, props: PagePropTypes): boolean {
+    const runId = this.getRunId();
+    // Always update highlighters for the first run
+    if (runId === 1) { return true; }
```

Comment:
This is for some reason required for the test (`scrolls to search result when selected before page navigation`) to pass(`Page.spec.tsx`)